server/src/main/java/io/pravega/schemaregistry/storage/client/WireCommandClient.java

```diff
@@ -0,0 +1,417 @@
+     */
+    CompletableFuture<Void> deleteTableSegment(final String tableName,
+                                               final boolean mustBeEmpty,
+                                               String delegationToken) {
+        return getTableUri(tableName)
+                .thenCompose(uri -> {
+                    final WireCommandType type = WireCommandType.DELETE_TABLE_SEGMENT;
+
+                    RawClient connection = new RawClient(ModelHelper.encode(uri), connectionPool);
+                    final long requestId = connection.getFlow().asLong();
+
+                    return sendRequest(connection, requestId, new WireCommands.DeleteTableSegment(requestId, tableName, mustBeEmpty, delegationToken))
+                            .thenAccept(rpl -> handleReply(rpl, connection, tableName, WireCommands.DeleteTableSegment.class, type));
+                });
+    }
+
+    /**
+     * This method sends a WireCommand to update table entries.
+     *
+     * @param tableName       Qualified table name.
+     * @param entries         List of {@link TableSegmentEntry} instances to be updated.
+     * @param delegationToken The token to be presented to the Segment Store.
+     * @return A CompletableFuture that, when completed normally, will contain the current versions of each
+     * {@link TableSegmentEntry}.
+     * If the operation failed, the future will be failed with the causing exception. If the exception can be retried
+     * then the future will be failed with {@link StoreExceptions}.
+     */
+    CompletableFuture<List<TableSegmentKeyVersion>> updateTableEntries(final String tableName,
+                                                                       final List<TableSegmentEntry> entries,
+                                                                       String delegationToken) {
+        return getTableUri(tableName).thenCompose(uri -> {
+            final WireCommandType type = WireCommandType.UPDATE_TABLE_ENTRIES;
+            List<Map.Entry<WireCommands.TableKey, WireCommands.TableValue>> wireCommandEntries = entries.stream().map(te -> {
+                final WireCommands.TableKey key = convertToWireCommand(te.getKey());
+                final WireCommands.TableValue value = new WireCommands.TableValue(te.getValue());
+                return new AbstractMap.SimpleImmutableEntry<>(key, value);
+            }).collect(Collectors.toList());
+
+            RawClient connection = new RawClient(ModelHelper.encode(uri), connectionPool);
+            final long requestId = connection.getFlow().asLong();
+            WireCommands.UpdateTableEntries request = new WireCommands.UpdateTableEntries(requestId, tableName, delegationToken,
+                    new WireCommands.TableEntries(wireCommandEntries), WireCommands.NULL_TABLE_SEGMENT_OFFSET);
+
+            return sendRequest(connection, requestId, request)
+                    .thenApply(rpl -> {
```

Comment:
Exception handling may be needed here too.

Reply:
same as above